src/cplus_plugin/gui/constant_raster_manager_dialog.py

```diff
@@ -0,0 +1,1135 @@
+# -*- coding: utf-8 -*-
+"""
+Dialog for managing constant rasters for NCS pathways and activities.
+"""
+
+import os
+import typing
+
+from qgis.PyQt import QtCore, QtGui, QtWidgets
+from qgis.core import (
+    QgsProject,
+    QgsRectangle,
+    QgsSettings,
+    QgsVectorLayer,
+    QgsRasterLayer,
+)
+from qgis.gui import QgsMessageBar
+from qgis.utils import iface
+
+from ..conf import settings_manager, Settings
+from ..models.base import ModelComponentType
+from ..models.constant_raster import (
+    ConstantRasterCollection,
+    ConstantRasterComponent,
+    ConstantRasterContext,
+    ConstantRasterMetadata,
+)
+from ..lib.constant_raster import (
+    constant_raster_registry,
+)
+from ..definitions.defaults import (
+    ICON_PATH,
+    YEARS_EXPERIENCE_PATHWAY_ID,
+    YEARS_EXPERIENCE_ACTIVITY_ID,
+)
+from .component_item_model import NcsPathwayItemModel, ActivityItemModel
+from .constant_raster_widgets import (
+    ConstantRasterWidgetInterface,
+    YearsExperienceWidget,
+)
+from ..utils import log
+
+
+class ConstantRastersManagerDialog(QtWidgets.QDialog):
+    """Dialog for managing constant rasters."""
+
+    # Signal emitted when user requests to create constant rasters
+    # Parameters: (context, collection, input_range, metadata_id, current_view)
+    # current_view: True = current view only, False = all constant raster types
+    create_rasters_requested = QtCore.pyqtSignal(object, object, tuple, str, bool)
```

Comment:
@osundwajeff perhaps better to use the actual class type instead of keeping it generic as `object`. Helps with readability.